File: src/api/src/StackForge.Application/MentorshipContext/UseCases/SearchMentorByStacks/SearchMentorByStacksResponse.cs

namespace StackForge.Application.MentorshipContext.UseCases.SearchMentorByStacks;

public sealed record SearchMentorByStacksResponse(
    Guid MentorId, 
    string FullName, 
    string CourseName, 
    string Institution, 
    IReadOnlyList<MentorStackResponse> Stacks
    );
    
public sealed record MentorStackResponse(Guid StackId, string Name, string Key);

File: src/web/stackforge-web/src/app/features/mentorship/models/SearchMentorByStacks/search-mentor-by-stacks.response.ts
export interface SearchMentorByStacksResponse {
  mentorId: string;
  fullName: string;
  courseName: string;
  institution: string;
  stacks: MentorStackResponse[];
}

export interface MentorStackResponse {
  stackId: string;
  name: string;
  key: string;
}